### PR TITLE
[DM-31633] Flesh out documentation

### PR DESCRIPTION
**Pull request by rra**
- Rename and reorganize sections based on co-work discussion
- Flesh out the troubleshooting section
- Add documentation for updating `TAP_SCHEMA`
- Add a first draft of documentation for service owners, greatly enhancing what was previously there

### CODE REVIEW

**Comment by rra** [comment]
https://phalanx.lsst.io/v/DM-31633/index.html has the published version of this PR.

**Comment by athornton** [comment]
I updated with my empty-parens documentation.

**Comment by jonathansick** [comment]
Some overall points:

- There's a page called "Create a new service" and another called "Add a new application to Phalanx". I'd suggest settling on either application or service (I think Frossie suggested service) so its not confusing.
- I think putting "Add a secret with 1Password and VaultSecret" _after_ "Add a new application to Phalanx" is a better flow.
- Make sure that the text in "Add a new application to Phalanx" links to "Add a secret with 1Password and VaultSecret".
- "General maintenance" — I know we decided on this but the terminology is super awkward since its really about development and operations.

**Review by cbanek — APPROVED** [review]
Looks great, thanks!

**Comment by rra** [comment]
@jonathansick, you have to add all of your secrets before you can add a new service to Phalanx, so I think the flow has to be in this order.Otherwise, the first line of the guide for adding a service has to say "first, go add your secrets following the guide in this other document" for a document that's listed later.

There was a lot of service vs. application confusion throughout. I did some global searches and replaces and I think terminology is now consistent.

I changed General mainttenance to General development and operations.

File: docs/app-guide/create-service.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+Select ``FastAPI application (Safir)`` from the list of project types.
+This will create a new GitHub repository with the basic framework of a FastAPI application that will work well inside the Rubin Science Platform.
+
+Any Python service destined for the RSP should regularly update its dependencies to pick up any security fixes.
+If your service follows the code layout of the FastAPI application template, using `neophile <https://neophile.lsst.io/>`__ to automatically create PRs to update your dependencies is strongly recommended.
+To add your service to the list of repositories that neophile updates, submit a PR to add the repository owner and name to `neophile's configuration <https://github.com/lsst-sqre/roundtable/blob/master/deployments/neophile/values.yaml>`__.
+
+Each release of your application should be tagged.
+The tag should use `semantic versioning`_ (for example, ``1.3.2``).
+Creating a GitHub release for the tag is optional but recommended, and we recommend setting the title of the release to the name of the tag.
```
[the remark this answers is not in the document]
Comment (rra) [reply docs/app-guide/create-service.rst]: I assume by "this" you mean the tags? I turned that into must. (The GitHub release doesn't seem to affect anything other than the display on GitHub.)

**Comment by jonathansick** [comment]
The reason I suggest that is because you're probably going to be in the midst of trying to deploy your app, and then realize that you need to manage secrets. So it makes sense for the first page to cover app development and the second page to cover app deployment, with the third page being a sub-task coming out of app deployment which is secrets management. For example, you don't necessarily know the name of your app before starting the process of app deployment. That's why I'd rather folks be reading the app deployment page first, and then being linked into the secrets page; contextually you're in a better place to understand what's going on.

**Comment by rra** [comment]
I'm not sure that I understand your mental model, since I'm not sure how you could not know the name of your service by the time you reached the add-service page. The first page is create-service where you create the repository, write the code, and write a Helm chart, and by that point you must have a name, no? We tell people to use the name of their Helm chart for their service and Argo CD application. And as part of doing that development, presumably you will figure out what secrets you need.

In my mental model, the steps are:

1. Write the application, which involves knowing what secrets your application needs
2. Write a Helm chart for that service, which includes vault-secret resources
3. Put the static secrets that you need into 1Password
4. Add the application to Phalanx, which includes adding the secret logic to the installer

The installer integration isn't documented much at the moment.

**Comment by jonathansick** [comment]
You're totally right. My mental model is/was that the app development is generic to the deployment and that in the first page of docs we wouldn't even be talking about Phalanx itself yet. But you're totally right that since we add things like the VaultSecret resource to our app's charts, rather than delaying putting it into a chart in Phalanx (akin to the kustomize overlays in lsst-sqre/roundtable), we're basically building services specifically with their Phalanx deployment in mind from the start. All good, sorry for the fuss.

**Pull request merged**